DAQ/src/CaloHitsFromFragments_module.cc

```diff
@@ -63,94 +74,81 @@ struct CrystalInfo {
   // --- Production:
   virtual void produce(Event&);
 
-
 private:
   void analyze_calorimeter_(const artdaq::Fragment& f,
                             std::unique_ptr<mu2e::CaloHitCollection> const& calo_hits,
-                            std::unique_ptr<mu2e::CaloHitCollection> const& caphri_hits
-			    );
+                            std::unique_ptr<mu2e::CaloHitCollection> const& caphri_hits);
 
   void addPulse(uint16_t& crystalID, float& time, float& eDep);
 
   int diagLevel_;
 
   art::InputTag caloFragmentsTag_;
-  float         digiSampling_;
-  float         deltaTPulses_, pulseRatioMax_, pulseRatioMin_;
+  float digiSampling_;
+  float deltaTPulses_, pulseRatioMax_, pulseRatioMin_;
 
-  const int     hexShiftPrint = 7;
+  const int hexShiftPrint = 7;
 
-  std::array<std::vector<CrystalInfo>, 674*4>  pulseMap_;
-  mu2e::CaloDAQUtilities                       caloDAQUtil_;
+  std::unordered_map<uint16_t, std::list<CrystalInfo>>
+      pulseMap_; // Temporary hack until the Calorimeter channel map is finialized
+  mu2e::CaloDAQUtilities caloDAQUtil_;
 
-  std::array<float, 674*4>                     peakADC2MeV_;
-  std::array<int, 4>                           caphriCrystalID_;
+  std::array<float, 674 * 4> peakADC2MeV_;
+  std::array<int, 4> caphriCrystalID_;
 };
 
-
 // ======================================================================
 
+void art::CaloHitsFromFragments::beginRun(art::Run& Run) {
 
-void art::CaloHitsFromFragments::beginRun(art::Run&  Run){
-
-  //FIX ME! 
+  // FIX ME!
   // here we need to load the prodition-service with the calibrations
-  // for converting teh peakADC into MeV. I decided to fill an array 
-  // with the conversion constants in order to speed up the access 
+  // for converting teh peakADC into MeV. I decided to fill an array
+  // with the conversion constants in order to speed up the access
   // NOW FILLING THE ARRAY WITH A DUMMY VALUE
-  for (size_t i=0; i<peakADC2MeV_.size(); ++i){
+  for (size_t i = 0; i < peakADC2MeV_.size(); ++i) {
     peakADC2MeV_[i] = 0.0461333;
   }
 
-  //FIX ME!
-  //the list of the SiPM-IDs of the channels that are used for the 
-  //calorimeter-lumi monitor should come from a pro-dition
-  caphriCrystalID_ = {1, 2, 3, 4};
- 
+  // FIX ME!
+  // the list of the SiPM-IDs of the channels that are used for the
+  // calorimeter-lumi monitor should come from a pro-dition
+  caphriCrystalID_ = {623, 624, 595, 596};
```

Comment:
For future reference these should come from Proditions or the geometry subsystem.  Ok for now.